torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -1081,6 +1082,18 @@ void TensorPipeAgent::join(bool shutdown) {
       // It is enough to wait for there to be no more active client calls, since
       // each server call corresponds to a client call for some other worker.
       callCountCV_.wait(lock, [this] { return clientActiveCalls_ == 0; });
+
+      // With dynamic RPC groups we are only concerned with active calls on
+      // local worker ActiveCallCount is 0 at this point and we will shutdown
+      // (any future calls will be dropped)
+      if (!isStaticGroup_) {
+        // Remove this agents worker info from store
+        removeCurrentName(rankToNameStore_, workerInfo_.id_, workerInfo_.name_);
+        if (shutdown) {
+          shuttingDown_ = true;
+        }
+        break;
+      }
```

Comment:
Shall we move this out of the while loop or create a dedicate function for it? Because IIUC, this is no longer a `join` and this does not need a loop?

Reply:
I moved it out of the loop and into a separate function but kept it in `join` since we should keep the same API of rpc calling `.shutdown()` which joins the agent and its' thread(s) with the main process.